packages/collabswarm/src/collabswarm-document.ts

```diff
@@ -110,58 +113,64 @@ export class CollabswarmDocument<
   }
 
   constructor(
-    /** */
+    /**
+     * Collabswarm swarm that this document belongs to.
+     */
     public readonly swarm: Collabswarm<
       DocType,
       ChangesType,
       ChangeFnType,
-      MessageType,
       PrivateKey,
       PublicKey,
       DocumentKey
     >,
-    /** */
+
+    /**
+     * Path of the document.
+     */
     public readonly documentPath: string,
-    /** */
+
+    /**
+     * CRDTProvider handles reading/writing CRDT document data and metadata.
+     */
     private readonly _crdtProvider: CRDTProvider<
       DocType,
       ChangesType,
-      ChangeFnType,
-      MessageType
+      ChangeFnType
     >,
-    /** */
+
+    /**
+     * AuthProvider handles signing/verification and encryption/decryption.
+     */
     private readonly _authProvider: AuthProvider<
       PrivateKey,
       PublicKey,
       DocumentKey
     >,
-    /** */
-    private readonly _changesSerializer: ChangesSerializer<ChangesType>,
-    /** */
-    private readonly _messageSerializer: MessageSerializer<MessageType>,
-  ) {}
 
-  /**
-   * Store new document key.
-   *
-   * @param documentKey: new key to use
-   *
-   * @remarks Safe to call multiple times since keys are stored in set without duplicates
-   */
-  public setDocumentKey(documentKey: DocumentKey): void {
-    this._currentDocumentKey = String(documentKey);
-    this._documentKeys.add(String(documentKey));
-  }
+    /**
+     * ACLProvider handles read/write ACL operations.
+     */
+    private readonly _aclProvider: ACLProvider<ChangesType, PublicKey>,
 
-  /**
-   * Get list of all document keys used to encrypt change messages
-   * used to decrypt change messages.
-   *
-   * @remark Since the document is created from change history, all keys are needed.
-   */
-  public getDocumentKeys(): Set<string> {
-    return this._documentKeys;
-  }
+    /**
+     * KeychainProvider handles read/write ACL operations.
+     */
+    private readonly _keychainProvider: KeychainProvider<
+      ChangesType,
+      DocumentKey
+    >,
+
+    /**
+     * ChangesSerializer is responsible for serializing/deserializing CRDTChangeBlocks.
+     */
+    private readonly _changesSerializer: ChangesSerializer<ChangesType>,
+
+    /**
+     * MessageSerializer is responsible for serializing/deserializing CRDTSyncMessages.
```

Comment:
The non-obvious part of the comment seems to be `CRDTSyncMessages` so maybe giving an example, or reminder of what that contains? (I searched the repo to give a suggestion, but don't find it defined yet...)

Reply:
Ah, CRDTSyncMessage is defined in `packages/collabswarm/src/crdt-sync-message.ts`. Perhaps the trailing `s` messed up a search?

Reply:
Perfect and well commented there 👍